src/Avalonia.Controls/Remote/RemoteServer.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Avalonia.Controls.Embedding;
+using Avalonia.Controls.Remote.Server;
+using Avalonia.Platform;
+using Avalonia.Remote.Protocol;
+
+namespace Avalonia.Controls.Remote
+{
+    public class RemoteServer
+    {
+        private EmbeddableControlRoot _topLevel;
+
+        class TopLevelImpl : RemoteServerTopLevelImpl, IEmbeddableWindowImpl
```

Comment:
Why is this class separate from `RemoteServerTopLevelImpl`?

Reply:
`RemoteServer` is a standalone class that can be used to host widgets for remoting. Previewer uses `RemoveServerTopLevelImpl` directly, since it needs more customization

Reply:
Couldn't we roll this `TopLevelImpl` class into `RemoteServerTopLevelImpl` anyway and use `RemoteServerTopLevelImpl` in `RemoteServer` and still allow the previewer to make customizations as needed?

Reply:
As you can see, `RemoteServerTopLevelImpl` only implements `ITopLevel`. `EmbeddableControlRoot` class requires `IEmbeddableWindowImpl` to be implemented. There is abosolutely no sence to implement that in `RemoteServerTopLevelImpl`, since it should only care about sizing, rendering and input.

Reply:
Can we rename `RemoteServerTopLevelImpl` then? I just don't like having `RemoteServerTopLevelImpl` and `RemoteServer.TopLevelImpl` as separate classes. It's confusing naming for the future imo. Maybe we could do `RemoteTopLevelImpl` with `RemoteServer.EmbeddableTopLevelImpl`? Just to clarify the separation between the two classes.